### ISSUE TITLE
Consider moving hosting to Netlify?

### ISSUE THREAD

**Issue by monsteronfire** [RapidRiverSoftware/rapidriversoftware.github.io#120]
I would like to explore the idea of hosting our site on Netlify instead of Github Pages. I think this would be a good idea because:

1. Netlify will allow us to use third party plugins
2. Sometimes the Jekyll version that’s compatible with Github Pages is older and it takes them a while to update it

The current workflow for deployment shouldn’t change.

## Further Context
I am currently working on a change that will allow us to save blog-post related images in the same folder as the markdown file. This sort of thing isn’t supported on Github Pages, because Jekyll recommends that images be put in the assets folder, so we currently reference images in posts like: 
```
/assets/posts/example-image-for-post.png
```
I don’t think this is the best approach, though. I think it would be easier if we were able to reference images like this:
```
example-image-for-post.png
```

I am open to discussing this further and seeing more opinions about it.

**Comment by massa1240** [comment]
As far as `/assets/posts/` is the convention, I don't see why not. The problem of having it in `assets` folder is that we might collide names... and to avoid that, we need to add the post slug in the folder and that would become something like `/assets/posts/name-of-your-post/example-image-for-post.png` and then I can see how that could become inconvenient, especially if we ever want to change this... isn't there some sort of helper that Jekyll allows us to add something like a prefix for images and add this any time we reference an image? Just throwing some ideas :p I feel like changing the host for this seems overkill haha but maybe that's the way to go

**Comment by calumhalcrow** [comment]
Thanks for the suggestion @monsteronfire. Some questions:

- How much would it cost us? Github Pages is free for us today.
- Outside of storing images for blog pages in the same directory as the blog post, are there other planned site features that would be easier/more convenient to implement as a result of this change? We would get the ability to use third-party plugins and have newer versions of Jekyll - would either of those help us with say the new Services page, Homepage and/or animation features?

**Comment by monsteronfire** [comment]
Thanks for the replies, @massa1240 and @calumhalcrow.

If we switch, we gain a better user experience when adding blog post images. What do we lose by leaving Github Pages?

I agree that it is a small gain, but I don't agree that it's overkill. I think we have to consider adding images to all our blog posts, so they aren't just massive walls of text. Improving the UX for a task like that isn't a bad thing, right?

Regarding your first question, Calum:
- For a blog our size, Netlify has a free tier, so it would be the same price as Github Pages

Here's a more comprehensive list of reasons why I feel Netlify might be a nicer solution (Warning: Incoming wall of text):
## # 1. Public vs Private Repo
- Currently, from what I understand, the free tier of Github Pages only allows you to use a public repo. In future, any draft blog posts we have will all be public as well. I just have personal issues with this, but it may not be a big deal for others
- Netlify allows you to deploy from a private repo

![Spider-Man-Pointing-Meme](https://user-images.githubusercontent.com/1836842/84731343-29a32100-afcb-11ea-90bc-06d4872bfd76.jpg)


## # 2. Asset Management
- Previously, I had a spike PR where I had added Webpack to our Jekyll site so we could optimise the JS and CSS. I wanted to improve the [page speed](https://developers.google.com/speed/pagespeed/insights/) score. I experimented with webpack, but for several reasons, that was not the right solution
- Netlify offers bundling and minification of assets. So we don't have to add Webpack just to get those features
- Netlify offers prerendering
- Out of the box, Netlify also provides image compression. So this further helps to speed things up


![Manager](https://user-images.githubusercontent.com/1836842/84731461-7f77c900-afcb-11ea-9bab-beeb7eb3a00e.JPG)



## # 3. Form Submissions
- We currently use formspree to handle our form submissions. I know it works fine, but Netlify offers form submissions out-of-the-box, so another small benefit would be that we don't need a separate service to handle forms

![messenger-pigeon](https://user-images.githubusercontent.com/1836842/84731895-dcc04a00-afcc-11ea-8981-f1b661dd09c2.jpg)



## # 4. The Jekyll Lock-in
- From what I can gather, using Github Pages means you have to use Jekyll. I think Jekyll is _fine_, but if we want to explore better options in future, we can't do that with Github Pages
- Netlify supports whatever static site generator. We could switch to Gatsby or even Hugo if we wanted to
- Gatsby would be an interesting possibility because we could use the company website as a tool for onboarding or training new members to use React. It definitely has a bunch of it's own issues, but right now, we don't even have the option to consider it
- I only mentioned Hugo because I am biased and think it's a superior tool for content sites

![Octopus_captured](https://user-images.githubusercontent.com/1836842/84731603-ebf2c800-afcb-11ea-9f0c-7478864d8f66.jpeg)



In summary, I think it's very apparent that I have a clear bias towards Netlify. This bias is probably just because I've used it a lot more extensively than Github Pages, so things I see as benefits may not be all that important. Which is why I wanted to start a discussion.

I look forward to seeing differing perspectives.

**Comment by calumhalcrow** [comment]
Okay, I'm not against the move per se.

I'm quite keen on improving the contact functionality. If it helps with that then so much the better.

How about we discuss/plan during the next Brand Weekly? There's going to be a level-of-effort associated with this and we should understand that before we start.

**Comment by monsteronfire** [comment]
@calumhalcrow Cool. Let's discuss it during the meeting. 👍

**Comment by massa1240** [comment]
that was an awesome research Shaii! I think all the other reasons you mentioned is worth the change because now we aren't talking about just images slug anymore but optmisations and other interesting stuff we could get with the change! The fact that github pages requires a public repo also bothers me... not that the information is private (and, well, it's available in the web anyway), but I see why we would leave draft private...

Anyway, that's something for you guys to debate... I am bouncing for now! And sorry for jumping in out of nowhere! Seems like I am a watcher in this repo and the topic interested me :grin: 

keep the nice work!

**Comment by monsteronfire** [comment]
Thank you for your input, @massa1240

**Comment by monsteronfire** [comment]
Closing this issue because we have moved to Netlify.

**Comment by calumhalcrow** [comment]
![i-see-what-you-did-there-rage-face-meme](https://user-images.githubusercontent.com/277504/89756939-76f8c600-db16-11ea-8329-9d7420a09e82.png)
